src/protocol/http/org/apache/jmeter/protocol/http/sampler/HTTPHC4Impl.java

```diff
@@ -1687,8 +1688,16 @@ private UrlEncodedFormEntity createUrlEncodedFormEntity(final String contentEnco
                 // Must decode the value now, so that when the
                 // httpclient encodes it, we end up with the same value
                 // as the user had entered.
-                parameterName = URLDecoder.decode(parameterName, urlContentEncoding);
-                parameterValue = URLDecoder.decode(parameterValue, urlContentEncoding);
+                try {
+                    URLCodec urlCodec = new URLCodec(contentEncoding);
+                    //parameterName = URLDecoder.decode(parameterName, urlContentEncoding);
+                    parameterName = urlCodec.decode(parameterName, urlContentEncoding);
+                    //parameterValue = URLDecoder.decode(parameterValue, urlContentEncoding);
+                    parameterValue = urlCodec.decode(parameterValue, urlContentEncoding);
+                } catch (DecoderException e) {
+                    log.error(contentEncoding + " exception in encoding!");
```

Comment:
When you add new log messages, try to use string formats. In this case it would be
``log.error("{} exception in encoding!", contentEncoding);``
I wonder, whether the message should include the parameter value and/or name to give more context.

Reply:
Review comments are incorporated